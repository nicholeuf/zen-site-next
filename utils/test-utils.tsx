import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import renderer from 'react-test-renderer';

import AppLayout from '@/components/AppLayout';
import GlobalStyles from '@/app/styles/GlobalStyles';

interface Props {
  children: React.ReactNode;
}

// Include header, main, and footer in rendered content
const MockLayout: React.FC<Props> = ({ children }) => {
  return <AppLayout>{children}</AppLayout>;
};

// Include just mui theme global styles
const MockStyles: React.FC<Props> = ({ children }) => {
  return <GlobalStyles>{children}</GlobalStyles>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockStyles, ...options });

const customRenderWithLayout = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockLayout, ...options });

const customSnapshotRender = (children: ReactElement) =>
  renderer.create(<MockStyles>{children}</MockStyles>);

const customSnapshotRenderWithLayout = (children: ReactElement) =>
  renderer.create(<MockLayout>{children}</MockLayout>);

export * from '@testing-library/react';
export {
  customRender as render,
  customRenderWithLayout as renderWithLayout,
  customSnapshotRender as renderSnapshot,
  customSnapshotRenderWithLayout as renderSnapshotWithLayout,
};
