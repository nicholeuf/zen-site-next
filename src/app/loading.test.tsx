/*
 * @jest-environment jsdom
 */
import { renderSnapshotWithLayout } from 'test-utils';

import Loading from './loading';

const mockUsePathname = jest.fn();

jest.mock('next/navigation', () => ({
  usePathname() {
    return mockUsePathname();
  },
}));

describe('The Loading Page', () => {
  beforeAll(() => {
    mockUsePathname.mockImplementation(() => '/contact');
  });

  test('has expected snapshot', async () => {
    const { container } = renderSnapshotWithLayout(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
