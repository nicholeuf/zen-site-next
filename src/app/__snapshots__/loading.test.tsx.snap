// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`The Loading Page has expected snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px,200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 1px,200px;
    stroke-dashoffset: -126px;
  }
}

.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  box-shadow: var(--Paper-shadow);
  background-image: var(--Paper-overlay);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  box-sizing: border-box;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: fixed;
  z-index: 1100;
  top: 0;
  left: auto;
  right: 0;
  --AppBar-background: #FA2742;
  --AppBar-color: #fff;
  background-color: var(--AppBar-background);
  color: var(--AppBar-color);
  background-color: #E8EAE3;
  border: 8px solid;
  border-color: #373833;
  min-height: 60px;
  height: 60px;
  font-weight: 700;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: none;
}

@media print {
  .emotion-0 {
    position: absolute;
  }
}

.emotion-1 {
  font-family: fontFamily;
  font-size: 30px;
  height: 100%;
  width: 60px;
  border-right: 8px solid;
  border-right-color: #373833;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1:hover {
  cursor: pointer;
}

.emotion-2 {
  margin: 0;
  font: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #373833;
  width: 100%;
  text-align: center;
}

.emotion-2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:hover {
  color: rgb(251, 82, 103);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-2::before {
  content: "nf";
}

.emotion-3 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0px;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  border: 0px solid;
}

.emotion-4 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-transition: -webkit-transform 0.25s ease;
  transition: transform 0.25s ease;
}

.emotion-5:hover {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.emotion-6 {
  margin: 0;
  font: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  letter-spacing: 1.25px;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #373833;
  -webkit-transition: color 0.25s ease;
  transition: color 0.25s ease;
}

.emotion-6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-6:hover {
  color: rgb(251, 82, 103);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-10 {
  margin: 0;
  font: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 4px solid;
  border-bottom-color: rgb(251, 82, 103);
  box-sizing: border-box;
  letter-spacing: 1.25px;
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #373833;
  -webkit-transition: color 0.25s ease;
  transition: color 0.25s ease;
}

.emotion-10:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10:hover {
  color: rgb(251, 82, 103);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-13 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  margin-top: 60px;
  min-height: calc(var(--viewport-height) - 60px - 175px);
  padding: 0!important;
  max-width: 100%!important;
}

@media (min-width:600px) {
  .emotion-13 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-13 {
    max-width: 1200px;
  }
}

.emotion-14 {
  min-height: calc(var(--viewport-height) - 60px - 175px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-15 {
  display: inline-block;
  -webkit-animation: animation-0 1.4s linear infinite;
  animation: animation-0 1.4s linear infinite;
  color: #FA2742;
}

.emotion-16 {
  display: block;
}

.emotion-17 {
  stroke: currentColor;
  stroke-dasharray: 80px,200px;
  stroke-dashoffset: 0;
  -webkit-animation: animation-1 1.4s ease-in-out infinite;
  animation: animation-1 1.4s ease-in-out infinite;
}

.emotion-18 {
  background-color: #373833;
  color: #E8EAE3;
  height: 175px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-19 {
  max-width: 600px;
  display: grid;
  gap: 4px;
  text-align: center;
  height: 100%;
  width: 100%;
}

@media (min-width:0px) {
  .emotion-19 {
    grid-template-areas: "nav" "madeby" "copyright" "source";
  }
}

@media (min-width:600px) {
  .emotion-19 {
    grid-template-areas: "nav madeby" "nav copyright" "nav source";
  }
}

.emotion-20 {
  grid-area: nav;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

@media (min-width:0px) {
  .emotion-20 {
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
  }
}

@media (min-width:600px) {
  .emotion-20 {
    -webkit-box-pack: start;
    -ms-flex-pack: start;
    -webkit-justify-content: flex-start;
    justify-content: flex-start;
  }
}

.emotion-21 {
  width: 25%;
  height: 100%;
}

.emotion-22 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 0px;
}

@media (min-width:0px) {
  .emotion-22 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:600px) {
  .emotion-22 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  padding: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: -webkit-transform 0.25s ease;
  transition: transform 0.25s ease;
}

@media (min-width:0px) {
  .emotion-23 {
    margin: 8px;
  }
}

@media (min-width:600px) {
  .emotion-23 {
    margin: 24px;
  }
}

.emotion-23:hover {
  -webkit-transform: scale(1.2);
  -moz-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.emotion-24 {
  margin: 0;
  font: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-24:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-25 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  --IconButton-hoverBg: rgba(0, 0, 0, 0.04);
  padding: 0px;
}

.emotion-25::-moz-focus-inner {
  border-style: none;
}

.emotion-25.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-25 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-25:hover {
  background-color: var(--IconButton-hoverBg);
}

@media (hover: none) {
  .emotion-25:hover {
    background-color: transparent;
  }
}

.emotion-25.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-25.MuiIconButton-loading {
  color: transparent;
}

.emotion-26 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  fill: currentColor;
  font-size: 1.5rem;
  color: #E8EAE3;
}

.emotion-31 {
  grid-area: madeby;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .emotion-31 {
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:600px) {
  .emotion-31 {
    -webkit-box-pack: end;
    -ms-flex-pack: end;
    -webkit-justify-content: flex-end;
    justify-content: flex-end;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

.emotion-32 {
  margin: 0;
  font-family: fontFamily;
  font-size: 26px;
}

@media (min-width:0px) {
  .emotion-32 {
    font-size: 24px;
  }
}

@media (min-width:600px) {
  .emotion-32 {
    font-size: 34px;
  }
}

.emotion-33 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  display: inline;
  color: #FA2742;
}

.emotion-34 {
  grid-area: copyright;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .emotion-34 {
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:600px) {
  .emotion-34 {
    -webkit-box-pack: end;
    -ms-flex-pack: end;
    -webkit-justify-content: flex-end;
    justify-content: flex-end;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

.emotion-35 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 2.66;
  text-transform: uppercase;
  text-transform: none;
}

.emotion-36 {
  grid-area: source;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) {
  .emotion-36 {
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (min-width:600px) {
  .emotion-36 {
    -webkit-box-pack: end;
    -ms-flex-pack: end;
    -webkit-justify-content: flex-end;
    justify-content: flex-end;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }
}

.emotion-37 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 2.66;
  text-transform: uppercase;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-38 {
  margin: 0;
  font: inherit;
  line-height: inherit;
  letter-spacing: inherit;
  color: #FA2742;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #E8EAE3;
}

.emotion-38:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-39 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: fontFamily;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border: 0;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding: 6px 8px;
  color: var(--variant-textColor);
  background-color: var(--variant-textBg);
  --variant-textColor: #FA2742;
  --variant-outlinedColor: #FA2742;
  --variant-outlinedBorder: rgba(250, 39, 66, 0.5);
  --variant-containedColor: #fff;
  --variant-containedBg: #FA2742;
  padding: 4px 5px;
  font-size: 0.8125rem;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: inherit;
  font-size: inherit;
  padding: 0px;
}

.emotion-39::-moz-focus-inner {
  border-style: none;
}

.emotion-39.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-39 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-39:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-39.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

@media (hover: hover) {
  .emotion-39:hover {
    --variant-containedBg: rgb(175, 27, 46);
    --variant-textBg: rgba(250, 39, 66, 0.04);
    --variant-outlinedBorder: #FA2742;
    --variant-outlinedBg: rgba(250, 39, 66, 0.04);
  }
}

.emotion-39.MuiButton-loading {
  color: transparent;
}

<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed emotion-0"
    data-testid="header"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root emotion-1"
      data-testid="header-logo"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-2"
        href="/"
      >
        <span
          class="MuiBox-root emotion-3"
          data-testid="visually-hidden"
        >
          Home
        </span>
      </a>
    </div>
    <nav
      aria-label="Main Navigation"
      data-testid="header-nav"
    >
      <ul
        class="MuiList-root MuiList-padding emotion-4"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-6"
            href="/work"
          >
            Work
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-6"
            href="/about"
          >
            About
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
        >
          <a
            aria-current="page"
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-10"
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-5"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-6"
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-13"
  >
    <div
      class="MuiBox-root emotion-14"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary emotion-15"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg emotion-16"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate emotion-17"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </main>
  <footer
    class="MuiBox-root emotion-18"
    data-testid="footer"
  >
    <div
      class="MuiBox-root emotion-19"
    >
      <div
        class="MuiBox-root emotion-20"
      >
        <nav
          aria-label="External Navigation"
          class="MuiBox-root emotion-21"
          data-testid="footer-nav"
        >
          <ul
            class="MuiList-root MuiList-padding emotion-22"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-23"
            >
              <a
                aria-label="LinkedIn (opens in new window)"
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-24"
                href="https://www.linkedin.com/in/nicholeuf"
                rel="noopener"
                target="_blank"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-25"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-26"
                    data-testid="LinkedInIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                    />
                  </svg>
                </button>
              </a>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding emotion-23"
            >
              <a
                aria-label="GitHub (opens in new window)"
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-24"
                href="https://github.com/nicholeuf"
                rel="noopener"
                target="_blank"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-25"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-26"
                    data-testid="GitHubIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                    />
                  </svg>
                </button>
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="MuiBox-root emotion-31"
      >
        <p
          class="MuiTypography-root MuiTypography-sacramento emotion-32"
        >
          Made with
           
          <span
            class="MuiTypography-root MuiTypography-body1 emotion-33"
          >
            ♥
          </span>
           
          by Nichole Frey
        </p>
      </div>
      <div
        class="MuiBox-root emotion-34"
      >
        <p
          class="MuiTypography-root MuiTypography-overline emotion-35"
        >
          Copyright © 
          2025
           -
           
          <strong>
            All Rights Reserved
          </strong>
        </p>
      </div>
      <div
        class="MuiBox-root emotion-36"
      >
        <p
          class="MuiTypography-root MuiTypography-overline emotion-37"
        >
          <a
            aria-label="View Source Code"
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover emotion-38"
            href="https://github.com/nicholeuf/zen-site-next"
            rel="noopener"
            target="_blank"
          >
            view source code
          </a>
           | 
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary emotion-39"
            tabindex="0"
            type="button"
          >
            View Credits
          </button>
        </p>
      </div>
    </div>
  </footer>
</div>
`;
